RadixWallet/Features/CreateAccount/Coordinator/CreateAccountCoordinator+Reducer.swift

```diff
@@ -112,33 +145,137 @@ extension CreateAccountCoordinator {
 
 	public func reduce(into state: inout State, childAction: ChildAction) -> Effect<Action> {
 		switch childAction {
-		case let .root(.step1_nameAccount(.delegate(.proceed(accountName, useLedgerAsFactorSource)))):
-			state.path.append(.step2_creationOfAccount(.init(
-				name: accountName,
-				networkID: state.config.specificNetworkID,
-				isCreatingLedgerAccount: useLedgerAsFactorSource
-			)))
+		case let .root(.nameAccount(.delegate(.proceed(accountName, useLedgerAsFactorSource)))):
+			state.name = accountName
+			if useLedgerAsFactorSource {
+				state.path.append(.selectLedger(.init(context: .createHardwareAccount)))
+				return .none
+			} else {
+				return .send(.internal(.derivePublicKeyFromDevice))
+			}
+
+		case let .path(.element(_, action: .selectLedger(.delegate(.choseLedger(ledger))))):
+			return .send(.internal(.derivePublicKeyFromLedger(ledger)))
+
+		case .path(.element(_, action: .completion(.delegate(.completed)))):
+			return .run { send in
+				await send(.delegate(.completed))
+				if isPresented {
+					await dismiss()
+				}
+			}
+
+		default:
+			return .none
+		}
+	}
+
+	public func reduce(into state: inout State, internalAction: InternalAction) -> Effect<Action> {
+		switch internalAction {
+		case .derivePublicKeyFromDevice:
+			state.destination = .derivePublicKey(
```

Comment:
Hmm, I guess you do have internal actions `derivePublicKeyFromDevice` and `derivePublicKeyFromLedger` so you have a single place to set the destination. But I'd say it is unnecessary to have additional actions that are pipped through TCA machinery, rather you can create a function which is called directly.